backend/src/apiserver/storage/resource_reference_store.go

```diff
@@ -64,6 +64,8 @@ func (s *ResourceReferenceStore) checkReferenceExist(tx *sql.Tx, referenceId str
 		selectBuilder = sq.Select("1").From("experiments").Where(sq.Eq{"uuid": referenceId})
 	case common.PipelineVersion:
 		selectBuilder = sq.Select("1").From("pipeline_versions").Where(sq.Eq{"uuid": referenceId})
+	case common.Namespace:
+		return true
```

Comment:
This simply checks whether there is the namespace resource exists or not. When runs are created, the request is transformed to a db schema. This function is called when the transformation happens.

In our case, we don't have a special table to store the namespace, thus always returning true. Since, there is no direct communication between profile and pipeline(e.g., when profile creates a new namespace, it notifies the KFP), I leave the relationship up to the user, especially when the user is authorized to create runs in this namespace.

Reply:
thanks for the explanation! sgtm

Reply:
could you explain this a little bit via comments?

Reply:
Done.